File: app/src/main/java/com/example/daypilot_supabase_test/DbTestScreen.kt
package com.example.daypilot_supabase_test

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import io.github.jan.supabase.auth.auth
import io.github.jan.supabase.auth.providers.builtin.Email
import io.github.jan.supabase.postgrest.postgrest
import kotlinx.coroutines.launch
import kotlinx.serialization.Serializable

@Serializable
data class TaskRow(
    val id: String? = null,
    val title: String = "",
    val category: String = "",
    val difficulty: String = "",
    val is_completed: Boolean = false
)

@Serializable
data class NewTask(
    val user_id: String,
    val title: String,
    val category: String = "General",
    val difficulty: String = "EASY"
)

@Composable
fun DbTestScreen() {
    var log by remember { mutableStateOf("Sin acciones todavía.") }
    val scope = rememberCoroutineScope()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp)
            .verticalScroll(rememberScrollState()),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Text("DayPilot — prueba Supabase", style = MaterialTheme.typography.titleLarge)

        Button(onClick = {
            scope.launch {
                log = try {
                    supabase.auth.signInWith(Email) {
                        email = "ana.garcia@daypilot.test"
                        password = "password123"
                    }
                    "Login OK. UID: ${supabase.auth.currentUserOrNull()?.id}"
                } catch (e: Exception) {
                    "Error login: ${e.message}"
                }
            }
        }) { Text("1. Login") }

        Button(onClick = {
            scope.launch {
                log = try {
                    val tasks = supabase.postgrest.from("tasks")
                        .select()
                        .decodeList<TaskRow>()
                    "Tareas: ${tasks.size}\n${tasks.joinToString("\n") { "- ${it.title}" }}"
                } catch (e: Exception) {
                    "Error tareas: ${e.message}"
                }
            }
        }) { Text("2. Traer tareas") }

        Button(onClick = {
            scope.launch {
                val uid = supabase.auth.currentUserOrNull()?.id
                log = if (uid == null) {
                    "Primero haz login"
                } else {
                    try {
                        supabase.postgrest.from("tasks")
                            .insert(NewTask(user_id = uid, title = "Tarea desde Android"))
                        "Tarea insertada OK"
                    } catch (e: Exception) {
                        "Error insertar: ${e.message}"
                    }
                }
            }
        }) { Text("3. Insertar tarea") }

        HorizontalDivider()
        Text("Resultado:", style = MaterialTheme.typography.titleMedium)
        Text(log)
    }
}

File: lib/db_test_screen.dart
import 'package:flutter/material.dart';
import 'package:supabase_flutter/supabase_flutter.dart';

final supabase = Supabase.instance.client;

class DbTestScreen extends StatefulWidget {
  const DbTestScreen({super.key});

  @override
  State<DbTestScreen> createState() => _DbTestScreenState();
}

class _DbTestScreenState extends State<DbTestScreen> {
  String _log = 'Sin acciones todavía.';
  bool _busy = false;

  Future<void> _run(Future<String> Function() action) async {
    setState(() => _busy = true);
    try {
      final result = await action();
      setState(() => _log = result);
    } catch (e) {
      setState(() => _log = 'Error: $e');
    } finally {
      setState(() => _busy = false);
    }
  }

  Future<String> _login() async {
    await supabase.auth.signInWithPassword(
      email: 'ana.garcia@daypilot.test',
      password: 'password123',
    );
    return 'Login OK. UID: ${supabase.auth.currentUser?.id}';
  }

  Future<String> _fetchTasks() async {
    final tasks = await supabase.from('tasks').select();
    final titles = tasks.map((t) => '- ${t['title']}').join('\n');
    return 'Tareas: ${tasks.length}\n$titles';
  }

  Future<String> _insertTask() async {
    final uid = supabase.auth.currentUser?.id;
    if (uid == null) return 'Primero haz login';
    await supabase.from('tasks').insert({
      'user_id': uid,
      'title': 'Tarea desde Flutter',
      'category': 'General',
      'difficulty': 'EASY',
    });
    return 'Tarea insertada OK';
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('DayPilot — prueba Supabase')),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            FilledButton(
              onPressed: _busy ? null : () => _run(_login),
              child: const Text('1. Login'),
            ),
            const SizedBox(height: 12),
            FilledButton(
              onPressed: _busy ? null : () => _run(_fetchTasks),
              child: const Text('2. Traer tareas'),
            ),
            const SizedBox(height: 12),
            FilledButton(
              onPressed: _busy ? null : () => _run(_insertTask),
              child: const Text('3. Insertar tarea'),
            ),
            const SizedBox(height: 24),
            const Divider(),
            const Text('Resultado:', style: TextStyle(fontWeight: FontWeight.bold)),
            const SizedBox(height: 8),
            if (_busy) const CircularProgressIndicator(),
            Text(_log),
          ],
        ),
      ),
    );
  }
}
